app/src/main/cpp/skyline/audio.cpp

```diff
@@ -0,0 +1,64 @@
+#include "audio.h"
+
+namespace skyline::audio {
+    Audio::Audio(const DeviceState &state) : state(state), oboe::AudioStreamCallback() {
+        oboe::AudioStreamBuilder builder;
+
+        builder.setChannelCount(constant::ChannelCount)
+            ->setSampleRate(constant::SampleRate)
+            ->setFormat(constant::PcmFormat)
+            ->setCallback(this)
+            ->openManagedStream(outputStream);
+
+        outputStream->requestStart();
+    }
+
+    std::shared_ptr<AudioTrack> Audio::OpenTrack(int channelCount, int sampleRate, const std::function<void()> &releaseCallback) {
```

Comment:
I would prefer to keep audiotrack abstracted away from the actual emulator. There may also be other uses where there can be more than just a signal call put in the callback.

Reply:
Any reason you want to keep it abstracted from the emulator because it is somewhat losing performance over just passing in a signal albeit tracks aren't opened too frequently so it's not exactly important.

Reply:
There is an optimization in audren we can do by updating the buffers on a release that I wouldn't mind leaving room open for and yeah latency isn't an issue for this either as audio is triple buffered.